build-scripts/rosa-sync/rosa_sync.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -xeo pipefail
+
+# This script copies the first AMI that is found to exist in the us-east-1 region in the meta.json
+# to the AWS account configured via the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables.
+# The path to the meta.json file must be passed as the first parameter.
+# An optional second parameter configures whether or not to do a dry run (defaults to false).
+ 
+error-exit() {
+    echo "Error: $*" >&2
+    exit 1
+}
+
+copy_ami() {
+    if [ "$#" -lt 1 ]; then
+        error-exit incorrect parameter count for copy_ami $#
+    fi
+
+    if ! [ -x "$(command -v aws)" ]; then
+        error-exit awscli command not found
+    fi
+
+    if [ -z "$AWS_ACCESS_KEY_ID" ]; then
+        error-exit \$AWS_ACCESS_KEY_ID is empty
+    fi
+    
+    if [ -z "$AWS_SECRET_ACCESS_KEY" ]; then
+        error-exit \$AWS_SECRET_ACCESS_KEY is empty
+    fi
+
+    # For ROSA, each released RHCOS AMI has to be synced to the us-east-1 region only.
+    # It will be replicated to other regions from there using the AWS Marketplace API.
+    local AWS_DEFAULT_REGION=${AWS_DEFAULT_REGION:-us-east-1}
+    local META_JSON_PATH="$1"
+    local DRY_RUN="$2"
+    local DRY_RUN_FLAG=""
+    if [ "$DRY_RUN" = true ]; then
+        DRY_RUN_FLAG="--dry-run"
+    fi
+
+    ami=$(jq -r "[.amis[]|select(.name==\"us-east-1\")][0].hvm" "${META_JSON_PATH}")
```

Comment:
this should be `jq -r "[.amis[\"us-east-1\"].hvm]" ...` based on my testing with the latest rhcos.json file here: https://github.com/openshift/installer/blob/master/data/data/rhcos.json

Reply:
When running manually, the input for this shouldn't be an installer `rhcos.json`!

In Jenkins, the URL to a `meta.json file` like this is passed to the job: https://releases-rhcos-art.cloud.privileged.psi.redhat.com/storage/releases/rhcos-4.6/46.82.202101131942-0/x86_64/meta.json (browser at https://releases-rhcos-art.cloud.privileged.psi.redhat.com/)
It's downloaded by the job and that path is later passed to `rosa_sync.sh` - so when running manually, a `meta.json` must be referenced.

The `meta.json` may contain multiple amis for one single region, which is why I sued the select pipe here.

Reply:
I've tested this script only with a manual dry run -- could you confirm it works as intended with a `meta.json`?

Reply:
We need @jharrington22 to weigh in here as I've been instructed to use the `rhcos.json` file instead to manually copy to the stage/prod rosa accounts. 
@jharrington22 - meta.json and rhcos.json are in different formats, what file do we want to use for the sync?

Reply:
@LorbusChris - I've confirmed it works with a meta.json manually and it's been copied the latest ami to the production rosa account. 
The meta.json used: https://releases-rhcos-art.cloud.privileged.psi.redhat.com/storage/releases/rhcos-4.6/46.82.202101150340-0/x86_64/meta.json

Reply:
thank you @dustman9000 for the confirmation, and my apologies to all involved for not raising awareness of the differences of formats here before.

The intended goal for this PR is to copy the AMI for each bootimage release over to ROSA.
It was decided that this pipeline (which updates the download links on the mirrors) is the best place to put it, since not all z-stream releases get a boot imagebump, but this particular pipeline is run manually for each bump by someone from ART who has a reference to the exact build to be released (it receives a particular build's `meta.json` as input).

The `rhcos.json` in the installer repo is also derivative of a build's `meta.json` and one should be able to backtrace it to its integral (https://github.com/openshift/installer/blob/master/hack/update-rhcos-bootimage.py#L2).